openprescribing/frontend/models.py

```diff
@@ -448,6 +458,112 @@ def current_version(self):
                 next_version = version.replaced_by
         return version
 
+    @cached_property
+    def dmd_objs(self):
+        '''Return highest level dm+d objects with this presentation's BNF code.
+        '''
+        for model in [VMP, AMP, VMPP, AMPP]:
+            objs = list(model.objects.filter(bnf_code=self.bnf_code))
+            if objs:
+                return objs
+        return []
+
+    @property
+    def dmd_obj_ids(self):
+        return [obj.id for obj in self.dmd_objs]
+
+    @property
+    def dmd_obj_type(self):
+        if not self.dmd_objs:
+            return None
+        else:
+            return type(self.dmd_objs[0])
+
+    def tariff_categories(self):
+        '''Return all tariff categories for this presentation.
+
+        Tariff categories are set at VMPP level, so this doesn't make sense for
+        presentations that correspond to AMPs.
+        '''
+
+        if self.dmd_obj_type == VMP:
+            filter_key = 'dtinfo__vmpp__vmp__id__in'
+        elif self.dmd_obj_type == VMPP:
+            filter_key = 'dtinfo__vmpp__id__in'
+        elif self.dmd_obj_type == AMPP:
+            filter_key = 'dtinfo__vmpp__ampp__id__in'
+        else:
+            return []
+
+        filter_params = {filter_key: self.dmd_obj_ids}
+        return DtPaymentCategory.objects.filter(**filter_params).distinct()
+
+    def tariff_categories_descr(self):
+        '''Return a description of the presentation's tariff category.'''
+        return ', '.join(tc.descr for tc in self.tariff_categories())
+
+    def availability_restrictions(self):
+        '''Return all availability restrictions for this presentation.
```

Comment:
"all availability restrictions for all members of this presentation's dm+d family"?